docs/extensions/constraints.md

```diff
@@ -0,0 +1,131 @@
+example, a repository's owners may choose to block force pushes to some
+branches, thus preventing the rewriting of history for that branch.
+Alternatively, a repository's owners may require all changes to some branch to
+receive at least two approvals, irrespective of the specific actors trusted,
+which can be declared through multiple levels of delegations.
+
+## Declaring Global Constraints
+
+This extension introduces the notion of "global constraints" (also referred to
+as "global rules") that are declared by the repository's owners in the root of
+trust metadata. Unlike the delegations implemented in gittuf today, global
+constraints can be of distinct types, with each type indicating how it is
+verified by a gittuf client. As with any changes to root of trust metadata,
+adding, removing, or updating a global constraint requires approval from a
+threshold of repository owners.
+
+## Verifying Global Constraints
+
+The gittuf verification workflow is extended to include the verification of all
+applicable global constraints after the standard workflow protecting the
+namespace being verified. The workflow used to verify a specific constraint
+depends on its type, and is discussed below alongside the proposed constraint
+types.
+
+## Global Constraint Schema
+
+Each global rule MUST have a name. This name MUST be unique among other global
+rules (irrespective of type), though the implementation may determine if the
+name must also be unique among all delegation names as well.
+
+Depending on the type of the global rule, additional information may be
+necessary. The extension defers the specific schema of global rules to the
+implementation, as different policy metadata versions may require different
+schemas.
+
+## Global Constraint Types
+
+This extension builds on the motivating examples and proposes two types of
+constraints. In time, more types may be added and these MUST be recorded in this
+extension document.
+
+### Block Force Pushes
+
+A frequently used security control for Git repositories is that which blocks
+force pushes to one or more branches. A force push results in the history of the
+branch being rewritten. Thus, the `block-force-pushes` global constraint
+prevents rewriting history for the specific Git references. As such, this
+contraint requires one or more patterns that identify the repository references
+protected by the constraint.
+
+This constraint type is verified for every entry for a reference protected by
+the constraint. When verifying an RSL reference entry, the previous unskipped
+RSL reference entry for the same reference is identified. To meet this
+constraint, the verification workflow determines if the current RSL reference
+entry's target commit is a descendent of the previous RSL reference entry's
+target commit. If yes, then the constraint is met. If not, then verification
+terminates with an error indicating the rule that was not met.
```

Comment:
This is starting to feel very tailored to exactly what you want to do in this one case.   Is a more general mechanism a good idea?

Reply:
It's a tough balance; each type checks something else altogether and must be defined separately. The general mechanism is:
a) the notion of global constraints
b) the ability to define new types of global constraints over time as we feel necessary

I'm not sure we can trivially generalize across different types of checks beyond this currently.